libs/portals/admin/ids-admin/src/screens/Client/components/AddPermissions/AddPermissions.tsx

```diff
@@ -0,0 +1,243 @@
+      // Populate the available tenants state for dropdown values
+      setAvailableTenants(availableTenants)
+
+      // Find the selected tenant based on the tenant router param
+      const selectedTenant = availableTenants.find((t) => t.id === tenant)
+
+      if (selectedTenant) {
+        setSelectedTenant(formatOption(selectedTenant, locale))
+        getAvailableScopesQuery({
+          variables: {
+            input: {
+              tenantId: selectedTenant.id,
+            },
+          },
+        })
+      }
+    }
+  }, [fetcher.data, locale])
+
+  useEffect(() => {
+    if (authAdminScopes?.data) {
+      // When scope data is queried we need to filter it for the selected environment
+      // and remove already added or allowed scopes.
+      setAvailableScopes(
+        authAdminScopes.data
+          .map((scope) =>
+            scope.environments.find((e) => e.environment === environment),
+          )
+          .filter(isDefined)
+          .filter(
+            (scope) =>
+              scope &&
+              ![...addedScopes, ...(allowedScopes ?? [])].find(
+                (s) => s.name === scope.name,
+              ),
+          ),
+      )
+    }
+  }, [authAdminScopes?.data])
+
+  // Add the selected scopes to the addedScopes array for
+  const handleAdd = () => {
+    // Add the new scopes to the permissions array
+    onAdd([...selected.values()])
+
+    // Reset the selected scopes
+    setSelected(new Map())
+
+    // Close the modal
+    onClose()
+  }
+
+  // Handle checkbox change
+  const onChange = (value: AuthAdminClientAllowedScope) => {
+    if (selected.has(value.name)) {
+      selected.delete(value.name)
+    } else {
+      selected.set(value.name, value)
+    }
+    setSelected(new Map(selected))
+  }
+
+  const handleTenantChange = (opt: Option) => {
+    setSelectedTenant(opt)
+
+    getAvailableScopesQuery({
+      variables: {
+        input: {
+          tenantId: opt.value as string,
+        },
+      },
+    })
+  }
+
+  return (
+    <Modal
+      label={formatMessage(m.permissionsModalTitle)}
+      title={formatMessage(m.permissionsModalTitle)}
+      id="add-permissions"
+      isVisible={isVisible}
+      onClose={onClose}
+      closeButtonLabel={formatMessage(m.closeModal)}
+      scrollType="inside"
+    >
+      <Box marginTop={1} marginBottom={4}>
+        <Text>{formatMessage(m.permissionsModalDescription)}</Text>
+      </Box>
+
+      <Box
+        marginBottom={3}
+        className={md ? styles.dropdownContainer : undefined}
+      >
+        <Select
+          name="tenant"
+          size="sm"
+          backgroundColor="blue"
+          label={formatMessage(m.tenant)}
+          onChange={(opt) => handleTenantChange(opt as Option)}
+          value={selectedTenant}
+          options={availableTenants.map((tenant) =>
+            formatOption(tenant, locale),
+          )}
+        />
+      </Box>
+
+      <ShadowBox flexShrink={1} overflow="auto">
+        <T.Table>
```

Comment:
Regarding modal dropdown cut off, I think an empty state or at lease min height on the table would fix it or at least make it better
```
        <T.Table style={{ minHeight: 300 }}>
```